README.md

```diff
@@ -97,13 +97,17 @@ local client = nakama.create_client(config)
 local session = nakama_session.restore()
 
 -- authenticate if no session was stored or if session has expired
-if not session or nakama_session.expired(session) then
+if not session or nakama_session.is_token_expired(session) or nakama.is_refresh_token_expired(session) then
```

Comment:
If the token is expired but the refresh token isn't expired, we want to do `nakama.session_refresh` rather than reauthenticate.

Reply:
Dang, you're right. I need to write some unit tests for this. I'll add a ticket. This should be right though?

```
if session and nakama_session.is_token_expired_soon(session) and not nakama.is_refresh_token_expired(session) then
    print("Session has expired or is about to expire. Refreshing.")
    session = nakama.session_refresh(client, session.refresh_token)
    nakama_session.store(session)
elseif not session or nakama_session.is_token_expired(session) then
    print("Session does not exist or it has expired. Must reauthenticate.")
    session = client.authenticate_email("bjorn@defold.se", "foobar123", nil, true, "britzl")
    nakama_session.store(session)
end
client.set_bearer_token(session.token)
```

Reply:
No, so the refresh token is longer-lived than the bearer token. The purpose is to use it to acquire a new bearer token without requiring input from the user. 

So the first condition looks right although you may want to check that the refresh token is nil or empty if it's the first re-auth on SDK update.

The second condition checks if the bearer token is expired, but I think it should check if the refresh token is expired.

Reply:
> No, so the refresh token is longer-lived than the bearer token

Good to know!

> you may want to check that the refresh token is nil or empty if it's the first re-auth on SDK update

Good point!

> The second condition checks if the bearer token is expired, but I think it should check if the refresh token is expired

Yes, you're right. I should probably save this for Monday morning and not do it Friday evening! :-)